client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-category-list.tsx

```diff
@@ -0,0 +1,129 @@
+import { Button } from '@automattic/components';
+import { __experimentalNavigatorBackButton as NavigatorBackButton } from '@wordpress/components';
+import { __experimentalUseFocusOutside as useFocusOutside } from '@wordpress/compose';
+import { Icon, chevronRight } from '@wordpress/icons';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useMemo, useState } from 'react';
+import { createPortal } from 'react-dom';
+import useCategoriesOrder from './hooks/use-categories-order';
+import NavigatorHeader from './navigator-header';
+import PatternListPanel from './pattern-list-panel';
+import { useSectionPatterns } from './patterns-data';
+import type { Pattern, Category } from './types';
+import './screen-category-list.scss';
+
+interface Props {
+	categories: Category[];
+	sectionsMapByCategory: { [ key: string ]: Pattern[] };
+	onDoneClick: () => void;
+	onSelect: (
+		type: string,
+		selectedPattern: Pattern | null,
+		categorySelected: string | null
+	) => void;
+	replacePatternMode: boolean;
+	selectedPattern: Pattern | null;
+	wrapperRef: React.RefObject< HTMLDivElement > | null;
+}
+
+const ScreenCategoryList = ( {
+	sectionsMapByCategory,
+	categories,
+	onDoneClick,
+	replacePatternMode,
+	onSelect,
+	selectedPattern,
+	wrapperRef,
+}: Props ) => {
+	const translate = useTranslate();
+	const [ categorySelected, setCategory ] = useState< string | null >( null );
+	const [ openPatternList, setOpenPatternList ] = useState< boolean | null >( null );
+	const sectionPatterns = useSectionPatterns();
+	const categoriesInOrder = useCategoriesOrder( categories );
+
+	const categoriesWithPatterns = useMemo( () => {
```

Comment:
Maybe we can simply render `null` if the category doesn't have any pattern, and then we're able to get rid of this memo. WDYT?

Reply:
Sure, that's better.